server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -431,6 +398,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder.endObject();
     }
 
+    protected boolean indexedByDefault() {
+        return true;
+    }
+
+    protected boolean docValuesByDefault() {
+        return true;
+    }
+
+    protected boolean storedByDefault() {
+        return false;
+    }
```

Comment:
Am I correct that these methods are only used for serialization? Should we also rely on these default for parsing?

Reply:
Unfortunately we can't use them for parsing because the builders/parsers are static.  I think builder parameters/parsing/serialization/merging can all be cleaned up a lot though; I have some ideas which I'll sketch out in a separate issue.